Server-side/src/auth/auth.guard.ts

```diff
@@ -0,0 +1,11 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest().header('Authorization');
+    console.log('request is ' + request + 'End');
+    if (!request) return false;
+    else return true;
+  }
```

Comment:
the current code only checks for the presence of the Authorization header but doesn't verify the token or extract the payload.

example in nest Js docs 
https://docs.nestjs.com/security/authentication